server/server.go

```diff
@@ -0,0 +1,173 @@
+package server
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"net"
+
+	ipfslite "github.com/hsanjuan/ipfs-lite"
+	"github.com/ipfs/go-cid"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	pb "textile.io/grpc-ipfs-lite/ipfs-lite"
+)
+
+type ipfsLiteServer struct {
+	pb.UnimplementedIpfsLiteServer
+
+	// should this be a file var or here?
+	peer *ipfslite.Peer
+}
+
+// StartServer starts the gRPC server
+func StartServer(peer *ipfslite.Peer, host string) error {
+	lis, err := net.Listen("tcp", host)
+	if err != nil {
+		return err
+	}
+
+	grpcServer := grpc.NewServer()
+	server := &ipfsLiteServer{
+		peer: peer,
+	}
+	pb.RegisterIpfsLiteServer(grpcServer, server)
+	err = grpcServer.Serve(lis)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *ipfsLiteServer) AddFile(ctx context.Context, req *pb.AddFileRequest) (*pb.AddFileResponse, error) {
+	reader := bytes.NewReader(req.GetData())
+	addParams := ipfslite.AddParams{
+		Layout:    req.AddParams.GetLayout(),
+		Chunker:   req.AddParams.GetChunker(),
+		RawLeaves: req.AddParams.GetRawLeaves(),
+		Hidden:    req.AddParams.GetHidden(),
+		Shard:     req.AddParams.GetShared(),
+		NoCopy:    req.AddParams.GetNoCopy(),
+		HashFun:   req.AddParams.GetHashFun(),
+	}
+	node, err := s.peer.AddFile(ctx, reader, &addParams)
+	if err != nil {
+		return nil, err
+	}
+
+	respBlock := pb.Block{
+		RawData: node.RawData(),
+		Cid:     node.Cid().String(),
+	}
+
+	respLinks := []*pb.Link{}
```

Comment:
Possible improvement:
```suggestion
 	respLinks :=make([]*pb.Link, len(node.Links()))
```
Which allocates the right size of the slice in advance instead of making grow on-demand while `append`ing to it. (slices grow twice the size everytime they get full). Making this change would mean also changing the `append` to be index-assigned: `respLinks[i] = &respLink` where `i` should be de ignored `range` var.

Reply:
Nice, making that change for sure